doc/react/model.md

```diff
@@ -0,0 +1,222 @@
+# hatchifyReact model
+
+    hatchifyReact has a sets of hooks and promises to be used depending on your app's preferences. Each set gives the ability to create, update, delete, and fetch records.
+
+- [Promises](#promises)
+  - [createOne](#createone)
+  - [deleteOne](#deleteone)
+  - [findAll](#findall)
+  - [findOne](#findone)
+  - [updateOne](#updateone)
+- [Hooks](#hooks)
+  - [useAll](#useall)
+  - [useCreateOne](#usecreateone)
+  - [useDeleteOne](#usedeleteone)
+  - [useOne](#useone)
+  - [useUpdateOne](#useupdateone)
+
+## Promises
+
+### createOne
+
+This async function returns a promise used to create a single record. The function accepts a `CreateType` object of attributes to be added. On success, the promise will return the record (`RecordType`).
+
+```tsx
+
+  const createTodo = async () => {
+    await hatchedReact.model.Todo.createOne({
+      id: "todo-id",
+      name: "Go Running",
+    }).then((res) => console.log(res))
+  }
+
+  ...
+
+  <button onClick={() => createTodo()}>Create Todo</button>
+
+```
+
+### deleteOne
+
+This async function returns a promise to delete a single record. It accepts a string equal to the id of the record to be fetched, On success, the promise will return `undefined`.
+
+```tsx
+
+  const deleteTodo = async () => {
+    await hatchedReact.model.Todo.deleteOne("todo-id")
+  }
+
+  ...
+
+  <button onClick={() => deleteTodo()}>Delete Todo</button>
+
+```
+
+### findAll
+
+This async function returns a promise to fetch all records. It accepts a query object used to filter data. Optional parameters in the object include: `include`, `fields`, `filter`, `sort`, and `page`. On success, the promise will return the records (Array of `RecordType`).
```

Comment:
There's the `react-rest` readme, but that only covers the hooks. If we want dedicated docs for those here it should probably be another PR (could probably use the same ticket), since this one is already pretty big.

Reply:
@nlundquist could I ask for your thoughts on the above? Should we have docs for non-hook CRUD methods?

Reply:
hey Luca, yeah we should definitely document these. We could break them off into separate documentation in a `rest-client-jsonapi` section and them link to them from here.

Reply:
Sounds good, I created a [shell ticket](https://bitovi.atlassian.net/browse/HATCH-518) for that. Resolving this thread, but let me know if you need anything else.